fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -1,23 +1,265 @@
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.Config;
+import com.starrocks.common.util.PropertyAnalyzer;
 import com.starrocks.qe.ConnectContext;
 import com.starrocks.sql.ast.AstVisitor;
+import com.starrocks.sql.ast.CreateMaterializedViewStatement;
+import com.starrocks.sql.ast.ExpressionPartitionDesc;
+import com.starrocks.sql.ast.QueryRelation;
+import com.starrocks.sql.ast.QueryStatement;
+import com.starrocks.sql.ast.SelectRelation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 public class MaterializedViewAnalyzer {
 
+    private static final Logger LOG = LoggerFactory.getLogger(MaterializedViewAnalyzer.class);
+
     public static void analyze(StatementBase stmt, ConnectContext session) {
         new MaterializedViewAnalyzerVisitor().visit(stmt, session);
     }
 
     static class MaterializedViewAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
+
+        @Override
+        public Void visitCreateMaterializedViewStatement(CreateMaterializedViewStatement statement,
+                                                         ConnectContext context) {
+            statement.getTableName().normalization(context);
+            QueryStatement queryStatement = statement.getQueryStatement();
+            ExpressionPartitionDesc expressionPartitionDesc = statement.getPartitionExpDesc();
+            //check query relation is select relation
+            if (!(queryStatement.getQueryRelation() instanceof SelectRelation)) {
+                throw new SemanticException("Materialized view query statement only support select");
+            }
+            SelectRelation selectRelation = ((SelectRelation) queryStatement.getQueryRelation());
+            // check alias except * and SlotRef
+            List<SelectListItem> selectListItems = selectRelation.getSelectList().getItems();
+            for (SelectListItem selectListItem : selectListItems) {
+                if (selectListItem.isStar()) {
+                    throw new SemanticException("Select * is not supported in materialized view");
+                } else if (!(selectListItem.getExpr() instanceof SlotRef)
+                        && selectListItem.getAlias() == null) {
+                    throw new SemanticException("Materialized view query statement select item " +
+                            selectListItem.getExpr().toSql() + " must has alias except base select item");
+                }
+            }
+            // analyze query statement, can check whether tables and columns exist in catalog
+            Analyzer.analyze(queryStatement, context);
+            // collect table from query statement
+            Map<TableName, Table> tableNameTableMap = AnalyzerUtils.collectAllTable(queryStatement);
+            tableNameTableMap.forEach((tableName, table) -> {
+                if (!tableName.getDb().equals(statement.getTableName().getDb())) {
```

Comment:
add a warning log and print the db name in the log

Reply:
Why we need add a log at here?

Reply:
add more informations in throw exception message